src/core/library.c

```diff
@@ -1932,6 +1933,34 @@ MsQuicClose(
     }
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+QUIC_API
+MsQuicCloseAsync(
+    _In_ _Pre_defensive_ const void* QuicApi,
+    _In_opt_ QUIC_CLOSE_COMPLETE_HANDLER Handler,
+    _In_opt_ void* Context
+    )
+{
+    if (QuicApi != NULL) {
+        QuicTraceLogVerbose(
+            LibraryMsQuicClose,
+            "[ api] MsQuicCloseAsync");
+        
+        // There's actually nothing to wait for here, so we can just free the API table
+        // and call the completion handler immediately.
+        CXPLAT_FREE(QuicApi, QUIC_POOL_API);
+        MsQuicRelease();
```

Comment:
`MsQuicRelease` calls `MsQuicLibraryUninitialize` which calls the following functions that _could_ block:

- `MsQuicRegistrationClose` for the stateless registration.
- `CxPlatWorkerPoolDelete` which waits on the `WorkerPool->Rundown`.

So, the code needs to handle these cases and return if those need to block, and then asynchronously continue execution when the blockers are finished.